encrypted-media/polyfill/edge-keystatuses.js

```diff
@@ -0,0 +1,144 @@
+(function() {
+
+    // This polyfill fixes the following problems with Edge browser
+    // (1) Various maplike methods for keystatuses are not supported or suported incorrectly
+    // (2) Key Ids exposed in keystatuses are incorrect (byte swaps)
+    if ( navigator.userAgent.toLowerCase().indexOf('edge') > -1 ) {
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        // The following function is the core of this JS patch. The rest of this file is infrastructure
+        // required to enable this function
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        function _proxyKeyStatusesChange( event ) {
+            this._keyStatuses.clear();
+            var keyStatuses = [];
+            this._session.keyStatuses.forEach( function( keyId, status ) {
+                var newKeyId = new Uint8Array( keyId );
+
+                function swap( arr, a, b ) { var t = arr[a]; arr[a] = arr[b]; arr[b] = t; }
+                swap( newKeyId, 0, 3 );
+                swap( newKeyId, 1, 2 );
+                swap( newKeyId, 4, 5 );
+                swap( newKeyId, 6, 7 );
+
+                keyStatuses.push( { key: newKeyId, status: status, ord: arrayBufferAsString( newKeyId ) } );
+            });
+
+            function lexicographical( a, b ) { return a < b ? -1 : a === b ? 0 : +1; }
```

Comment:
The whole section, or a specific part ?

The whole section iterates through the key ids exposed in the native keyStatuses map. For each key id, we make a copy, fix byte order within the key id and add the new key id and its status for the local `keyStatuses` var.

Then we sort that array in lexicographical order of (fixed) key id and add the keys to the `_keyStatuses` map on our proxy object. The iteration order of a JS Map is the insertion order, so this ensures we get the correct iteration order.